### PR TITLE
feat(network intercept): populate "intercepts" in base event params

### PR DESCRIPTION
**Pull request by thiagowfx**
Bug: https://github.com/GoogleChromeLabs/chromium-bidi/issues/644

### CODE REVIEW

File: src/bidiMapper/domains/network/NetworkStorage.ts [hunk 1]
```diff
@@ -291,32 +287,14 @@ export class NetworkStorage {
...
-    requestId: Network.Request
+    requestId: Network.Request,
+    phase: Network.InterceptPhase
   ): Network.Intercept[] {
     const request = this.#requestMap.get(requestId);
     if (!request) {
       return [];
     }
 
-    let phase: Network.InterceptPhase | undefined = undefined;
```
Comment (thiagowfx) [inline src/bidiMapper/domains/network/NetworkStorage.ts:305]: This wasn't being used for anything else. It was just a literal implementation of https://w3c.github.io/webdriver-bidi/#network-intercepts

File: src/bidiMapper/domains/network/NetworkRequest.ts [hunk 2]
```diff
@@ -86,16 +85,20 @@ export class NetworkRequest {
...
     requestId: Network.Request,
-    eventManager: EventManager,
+    networkStorage: NetworkStorage,
     cdpTarget: CdpTarget,
     redirectCount = 0
   ) {
-    this.requestId = requestId;
-    this.#eventManager = eventManager;
+    this.#requestId = requestId;
+    this.#networkStorage = networkStorage;
```
Comment (Lightning00Blade) [inline src/bidiMapper/domains/network/NetworkRequest.ts:93]: Please inject both `networkStorage` and `eventManager` separately, I look at it and we should not have it on the networkStorage at all.

File: src/bidiMapper/domains/network/NetworkRequest.ts [hunk 3]
```diff
@@ -394,14 +397,24 @@ export class NetworkRequest {
   }
 
   #getBaseEventParams(phase?: Network.InterceptPhase): Network.BaseParameters {
+    const isBlocked = phase !== undefined && phase === this.#interceptPhase; // TODO: Set this in terms of intercepts?
```
Comment (Lightning00Blade) [inline src/bidiMapper/domains/network/NetworkRequest.ts:400]: Nit: add the comment above.

File: src/bidiMapper/domains/network/NetworkRequest.ts [hunk 4]
```diff
@@ -394,14 +397,24 @@ export class NetworkRequest {
...
     return {
-      isBlocked: phase !== undefined && phase === this.#interceptPhase,
+      isBlocked,
       context: this.#context,
       navigation: this.#getNavigationId(),
       redirectCount: this.#redirectCount,
       request: this.#getRequestData(),
       // Timestamp should be in milliseconds, while CDP provides it in seconds.
       timestamp: Math.round((this.#request.info?.wallTime ?? 0) * 1000),
+      ...(isBlocked
```
[the remark this answers is not in the document]
Comment (Lightning00Blade) [reply src/bidiMapper/domains/network/NetworkRequest.ts:410]: We JSON.strigify so this gets ignored if its `undefined`.

**Review by Lightning00Blade — APPROVED** [review]

File: src/bidiMapper/domains/network/NetworkRequest.ts (hunk 2 above)
Comment (thiagowfx) [reply src/bidiMapper/domains/network/NetworkRequest.ts:93]: Done

**Review by OrKoN — APPROVED** [review]

**Review by Lightning00Blade — APPROVED** [review]

**Comment by thiagowfx** [comment]
Updated E2E to add "intercepts"

**Pull request merged**